### PR TITLE
Modify kindtest.sh to choose right profile to run individual tests

### PR DESCRIPTION
**Pull request by anpanigr**
With this change, no need to change the maven profile to integration-tests while running individual tests.
The kind script and Jenkinfile  will automatically override the profile to integration-tests

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12115/ (single test with default)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12118/ (all test and integration-tests) (-)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12119/ (all test and default )

https://build.weblogick8s.org:8443/job/WebLogic%20Kubernetes%20Operator%20Kind/view/change-requests/job/PR-3333/6

### CODE REVIEW

File: Jenkinsfile.kindnightly [hunk 1]
```diff
@@ -481,7 +478,10 @@ EOF
                             if [ "${MAVEN_PROFILE_NAME}" == "kind-sequential" ]; then
                                 echo 'Overriding the PARALLEL_RUN to false for kind-sequential profile'
                                 PARALLEL_RUN=false
-                            elif [ ! -z "${IT_TEST}" ]; then
+                            fi
+                            if [ ! -z "${IT_TEST}" ]; then
+                                echo 'Overriding the MAVEN_PROFILE_NAME to integration-test'
```
Comment (vanajamukkara) [inline Jenkinsfile.kindnightly:483]:
```suggestion
                                echo 'Overriding MAVEN_PROFILE_NAME to integration-test when running individual test(s)`
```

File: kindtest.sh [hunk 2]
```diff
@@ -287,33 +287,31 @@ docker ps
...
-fi
-
 # Check for invalid Test Filter/Maven Profile Combination
-
-# Test Filter (all) and Maven Profile is integration-tests
-if [ "x${maven_profile_name}" = "xintegration-tests" ] && 
+if [ "${maven_profile_name}" == "integration-tests" ] && 
    [ "${test_filter}" = "**/It*" ] ; then
-     echo '(ERROR) All tests cannot be run with [integration-tests] profile'
+     echo "(ERROR) All tests cannot be run with [integration-tests] profile"
```
Comment (vanajamukkara) [inline kindtest.sh:293]:
```suggestion
     echo "(ERROR) All tests cannot be run with [integration-tests] profile, choose different profile"
```

File: kindtest.sh [hunk 3]
```diff
@@ -287,33 +287,31 @@ docker ps
...
-   [ "${test_filter}" != "**/It*" ] ; then
-    echo '(ERROR) Individual Test MUST be run with [integration-tests] profile'
-    exit 0
+if [ "${maven_profile_name}" == "kind-sequential" ]; then
+   echo "Overriding the parallel_run to false for kind-sequential profiler"
+   parallel_run=false
+   threads=1
 fi
 
+# If the IT_TEST is not set to "**/It*" (i.e. individual tests are chosen),
```
Comment (vanajamukkara) [inline kindtest.sh:303]: If IT_TEST is set, integration-test profile is used to run the tests. MAVEN_PROFILE_NAME parameter is ignored.

File: Jenkinsfile.kindnightly (hunk 1 above)
Comment (anpanigr) [reply Jenkinsfile.kindnightly]: done

File: kindtest.sh (hunk 2 above)
Comment (anpanigr) [reply kindtest.sh]: done

File: kindtest.sh (hunk 3 above)
Comment (anpanigr) [reply kindtest.sh]: done

**Review by vanajamukkara — APPROVED** [review]
LG

**Review by maggiehe00 — APPROVED** [review]
lgtm

**Pull request merged**